src/main/java/labaSolvd/JacoLuna/Services/entityServices/ReviewsService.java

```diff
@@ -49,34 +67,196 @@ public void add() {
+                    else
+                        reviewList = JsonParser.unparseToList(Review.class, JsonPaths.REVIEWS);
+
+                    if (reviewList != null && !reviewList.isEmpty()) {
+                        reviewList.remove(reviewList.stream()
+                                .filter(p -> p.getIdReview() == id)
+                                .findFirst()
+                                .orElse(null)
+                        );
+                    }
+                    if (source == SourceOptions.XML) {
+                        this.reviews.setReviews(reviewList);
+                        Marshaller.MarshallList(this.reviews, Reviews.class, XmlPaths.REVIEWS);
+                    } else
+                        JsonParser.parse(reviewList, JsonPaths.REVIEWS);
+                } catch (JAXBException e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+            case DATA_BASE -> {
+                try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+                    if (session.getMapper(ReviewMapper.class).deleteReview(id) > 0) {
+                        session.commit();
+                        result = true;
+                    }
+
+                } catch (Exception e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+        }
+        return result;
     }
 
     @Override
     public List<Review> getAll() {
-        if (source == SourceOptions.DATA_BASE)
-//            return reviewDAO.getList();
-            return null;
-        try {
-            return Marshaller.UnMarshallList(Reviews.class, XmlPaths.REVIEWS);
-        } catch (JAXBException e) {
-            Utils.CONSOLE_ERROR.error(e);
+        switch (source) {
+            case XML -> {
+                try {
+                    return Marshaller.UnMarshallList(Reviews.class, XmlPaths.REVIEWS);
+                } catch (JAXBException e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+            case JSON -> {
+                return JsonParser.unparseToList(Review.class, JsonPaths.REVIEWS);
+            }
+            case DATA_BASE -> {
+                try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+                    return session.getMapper(ReviewMapper.class).getAllReviews();
+                } catch (Exception e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
         }
         return null;
     }
 
     @Override
     public List<Review> search() {
-        return List.of();
+        List<Review> searchList = null;
+        try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+            List<Field> attributes = Arrays.stream(Review.class.getDeclaredFields()).toList();
+            int attIndex = selectAtt(attributes);
```

Comment:
Instead of calling multiple times `attributes.get(attIndex)` (Lines 175, 177, 179, 185), store the Field and access it directly.
```suggestion
            Field att = attributes.get(selectAtt(attributes));
```